### PR TITLE
GMA UI Tests Workflow

### PR DESCRIPTION
**Pull request by sunmou99**
### Description
1. Modified the GMA tests. Some of them are not UI Tests. Disabled TestAdInspector.
2. Slightly modified the UI Test framework. Will filter out non UI Testapps.
3. Added UI Test in integration_test workflow.
***

[replace this line]: # (Describe your testing in detail.)
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [x] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

**Comment by DellaBitta** [comment]
Why is the TestAdInspector test disabled / commented out?

**Comment by sunmou99** [comment]
> Why is the TestAdInspector test disabled / commented out?

We agreed not to include TestAdInspector in to our [UI Tests](https://docs.google.com/document/d/1a01c5W4U26tataczYw-sv5k_YsfVy4DBhuj_CdhEF8M/edit?resourcekey=0-qAIEiUCh4uHJoxCQZjgZnA#heading=h.cg1leil5anlr). 
Do you want to added it back?

**Comment by DellaBitta** [comment]
Yes, it should still be part of manual testing.

**Review by DellaBitta — CHANGES_REQUESTED** [review]

**Comment by sunmou99** [comment]
> Yes, it should still be part of manual testing.

Added back:)

File: gma/integration_test/src/integration_test.cc [hunk 1]
```diff
@@ -980,7 +982,7 @@ TEST_F(FirebaseGmaUITest, TestRewardedAdLoadAndShow) {
   LogDebug("Waiting for a moment to ensure all callbacks are recorded.");
   app_framework::ProcessEvents(2000);
 
-  EXPECT_EQ(content_listener.num_ad_clicked(), 1);
+  // EXPECT_EQ(content_listener.num_ad_clicked(), 1);
```
[the remark this answers is not in the document]
Comment (DellaBitta) [reply gma/integration_test/src/integration_test.cc:985]: Is there a way to conditionally check if this is being manually tested (or rather, not automatically tested) and add the check in those scenarios?

Failing that a big noticeable log of "Received %d num_ad_clicked events".

[the remark this answers is not in the document]
Comment (sunmou99) [reply gma/integration_test/src/integration_test.cc:985]: It's possible but the code may look weird. Let me add some comment as well.

**Review by DellaBitta — APPROVED** [review]

**Pull request merged**